src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/GroupResponsibilityController.kt

```diff
@@ -68,14 +69,15 @@ class GroupResponsibilityController(
         context.addAttachmentHeader(getTemplateName(type, path, expectationMapping))
 
         val expectations = expectationMapping.expectation
-        val rowCount = expectations.expectedRows().count()
         return if (type == "central")
         {
-            EmptyDataTable.new<BurdenEstimate>(rowCount, expectations.outcomes)
+            FlexibleDataTable.new(expectations.expectedCentralRows(expectationMapping.disease),
+                    expectations.outcomes, NullToEmptyStringSerializer.instance)
         }
         else
         {
-            EmptyDataTable.new<StochasticBurdenEstimate>(rowCount, expectations.outcomes)
+            FlexibleDataTable.new(expectations.expectedStochasticRows(expectationMapping.disease),
+                    expectations.outcomes, NullToEmptyStringSerializer.instance)
```

Comment:
Again, seems like more of this code could be shared between the two paths? In fact, could `type` just be an argument to a single method on the `Expectations` object?

Reply:
It could be, yes. See comment in `webmodels`. If you feel strongly I'll change, I just thought it was clearer this way - I think there's something a bit off about returning different types of things depending on a method argument!